### PR TITLE
Rc bon fix probable highest nonce

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed problem when nodes are entered in sync mode because of a probable highest nonce set by a proposed block

### CODE REVIEW

File: consensus/spos/worker.go [hunk 1]
```diff
@@ -267,21 +267,24 @@ func (wrk *Worker) ProcessReceivedMessage(message p2p.MessageP2P, _ func(buffToS
 
 		//TODO: Block validity should be checked here and also on interceptors side, taking into consideration the following:
 		//(previous random seed, round, shard id and current random seed to verify if the block has been sent by the right proposer)
-		isHeaderValid := !check.IfNil(header) && headerHash != nil
-		if isHeaderValid {
-			errNotCritical := wrk.forkDetector.AddHeader(header, headerHash, process.BHProposed, nil, nil, false)
-			if errNotCritical != nil {
-				log.Debug("add header in forkdetector", "error", errNotCritical.Error())
-			}
+		isHeaderInvalid := check.IfNil(header) || headerHash == nil
```
Comment (iulianpascalau) [inline consensus/spos/worker.go]: This is also done in PR #760 :D

File: consensus/spos/worker.go [hunk 2]
```diff
@@ -267,21 +267,24 @@ func (wrk *Worker) ProcessReceivedMessage(message p2p.MessageP2P, _ func(buffToS
...
+		}
 
-			log.Debug("received proposed block",
-				"from", core.GetTrimmedPk(core.ToHex(cnsDta.PubKey)),
-				"header hash", cnsDta.BlockHeaderHash,
-				"round", header.GetRound(),
-				"nonce", header.GetNonce(),
-				"prev hash", header.GetPrevHash(),
-			)
+		err = wrk.forkDetector.AddHeader(header, headerHash, process.BHProposed, nil, nil, false)
```
Comment (iulianpascalau) [inline consensus/spos/worker.go]: Why do we still add the proposed block if it is not been taken into account? (you removed lines from 168:171) from baseForkDetector? Might also remove process.BHProposed constant

File: process/sync/baseForkDetector.go [hunk 3]
```diff
@@ -165,10 +164,6 @@ func (bfd *baseForkDetector) removeCheckpointsBehindNonce(nonce uint64) {
 // computeProbableHighestNonce computes the probable highest nonce from the valid received/processed headers
 func (bfd *baseForkDetector) computeProbableHighestNonce() uint64 {
 	probableHighestNonce := bfd.finalCheckpoint().nonce
-	lastProposedBlockNonce := bfd.lastProposedBlockNonce()
```
Comment (iulianpascalau) [inline process/sync/baseForkDetector.go]: Closing security gaps. :+1:

File: consensus/spos/worker.go (hunk 2 above)
Comment (SebastianMarian) [reply consensus/spos/worker.go]: It is still used as a trigger for forced fork when consensus is stuck and cannot advance after 20 proposed blocks

File: consensus/spos/worker.go (hunk 1 above)
Comment (SebastianMarian) [reply consensus/spos/worker.go]: Ok. It would be merged then

File: consensus/spos/worker.go (hunk 2 above)
Comment (iulianpascalau) [reply consensus/spos/worker.go]: ok. great

**Pull request merged**